src/main/java/org/opentripplanner/service/vehiclerental/street/Composite.java

```diff
@@ -0,0 +1,97 @@
+package org.opentripplanner.service.vehiclerental.street;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.opentripplanner.street.model.RentalRestrictionExtension;
+import org.opentripplanner.street.search.state.State;
+
+/**
+ * Combines multiple restrictions into one.
+ */
+public final class Composite implements RentalRestrictionExtension {
```

Comment:
This class is public, so it need a better name, e.g. `CompositeRentalRestrictionExtension`.